projects/plugins/jetpack/extensions/blocks/videopress/v6/edit.js

```diff
@@ -39,16 +39,36 @@ export default function VideoPressEdit( { attributes, setAttributes } ) {
 		controls,
 	} );
 
-	// Uploading file to backend states.
-	const [ uploadingProgress, setUploadingProgress ] = useState( [] );
+	/*
+	 * Tracking state when uploading the video file.
+	 * uploadingProgress is an array with two items:
+	 *  - the first item is the upload progress
+	 *  - the second item is total
+	 */
+	const [ uploadingProgress, setUploadingProgressState ] = useState( [] );
+
+	/*
+	 * Uploading progress handler, wrapper by the useCallback() hook
+	 * to avoid re-rendering the component when the progress is updated.
+	 */
+	const setUploadingProgress = useCallback( function ( ...args ) {
+		setUploadingProgressState( args );
+	}, [] );
+
+	/*
+	 * It's considered the file is uploading
+	 * when the progress value is lower than the total.
+	 */
 	const isUploadingFile = !! (
-		uploadingProgress?.length && uploadingProgress[ 0 ] !== uploadingProgress[ 1 ]
+		uploadingProgress?.length && uploadingProgress[ 0 ] < uploadingProgress[ 1 ]
```

Comment:
You could extract both values to `const` and just reuse them.

```javascript
// no idea about the const names, just example
const zero = uploadingProgress?.[ 0 ]
const one = uploadingProgress?.[ 1 ]
```

```javascript
const isUploadingFile = !! ( zero < one )
const fileHasBeenUploaded = !! ( zero === one )
```

Reply:
Ok. Up to you 👍🏽  ... I prefer it in terms of readability and organization and you still don't need to use `.length` for it.

Reply:
Also, this code will define `fileHasBeenUploaded` is `true` when even `uploadingProgress` is an empty away (initial state)